src/herder/TransactionQueue.cpp

```diff
@@ -53,47 +79,146 @@ TransactionQueue::tryAdd(TransactionFramePtr tx)
         return TransactionQueue::AddResult::ADD_STATUS_TRY_AGAIN_LATER;
     }
 
-    auto info = getAccountTransactionQueueInfo(tx->getSourceID());
+    int64_t netFee = tx->getFeeBid();
+    int64_t seqNum = 0;
+
+    stateIter = mAccountStates.find(tx->getSourceID());
+    if (stateIter != mAccountStates.end() &&
+        !stateIter->second.mTransactions.empty())
+    {
+        auto& transactions = stateIter->second.mTransactions;
+
+        if (tx->getEnvelope().type() != ENVELOPE_TYPE_TX_FEE_BUMP)
+        {
+            seqNum = transactions.back()->getSeqNum();
+            oldTxIter = transactions.end();
+        }
+        else
+        {
```

Comment:
we should probably have a comment that just says "replace by fee" as the code is non trivial at a glance